### ISSUE TITLE
Validation values are stale

### ISSUE THREAD

**Issue by warrenday** [jannikbuschke/formik-antd#102]
When using fields such as "TimePicker" or "DatePicker". The values passed into Formiks "validate" or "validateSchema" are incorrect. The callback is showing stale values.

This is only true when validation is called after an `onBlur` event. `onChange` works fine.

```
 <Formik
      initialValues={initialValues}
      validate={values => {
        console.log(values); // values are stale :(
      }}
      onSubmit={onSubmit}
      validateOnChange={false}
      validateOnBlur={true}
    >
    ...
```

```
 <Formik
      initialValues={initialValues}
      validate={values => {
        console.log(values); // values are fresh :)
      }}
      onSubmit={onSubmit}
      validateOnChange={true}
      validateOnBlur={false}
    >
    ...
```

```
 <Formik
      initialValues={initialValues}
      validate={values => {
        console.log(values); // callback is fired twice, once with fresh values, and once with stale values
      }}
      onSubmit={onSubmit}
      validateOnChange={true}
      validateOnBlur={true}
    >
    ...
```

**Comment by nimiak** [comment]
This is related to jaredpalmer/formik#2083.

Works fine with formik v1.5.8

**Comment by jannikbuschke** [comment]
@warrenday is this still an issue with formik v2.0.8?

**Comment by ericdvb** [comment]
I'm seeing this issue after an upgrade to latest (2.0.8). I'm using a custom field component (at the bottom however, it's just an `input` element), and the behavior is the same - the validation callback is fired once with the updated value, and then a second time with the last / stale value. Happy to provide more details and would love to know if this issue is caused by something about the way I've configured my Formik instance or fields.

**Comment by ericdvb** [comment]
@warrenday I'm seeing this behavior as well after upgrading to latest Formik. Can't speak to the specific way you have your form / field set up, but in our case in v1.5.2 (previous version our project was on) the validation fires twice on change, the second time with the fresh value. In v2.0.8, it fires twice on change, the second time using the stale value.

It also seems that setting `validateOnChange` to false doesn't have the expected effect (validation fires on change anyway, but only once, and with the stale value) 😕 but that's a separate issue.

**Comment by jannikbuschke** [comment]
this should be fixed in v1.6.3 and v1.7.0-beta.1
